### ISSUE TITLE
Empty commit message when force-pushing

### ISSUE THREAD

**Issue by qwerty287** [woodpecker-ci/woodpecker#749]
### Component

web-ui

### Describe the bug

The commit message is empty after a force push which does not add new commits. It should view the latest commit's message.

### System Info

```json
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"next-85d70266"}
```

### Additional context

![Screenshot_20220205-193706](https://user-images.githubusercontent.com/80460567/152654834-e38912ee-c598-4899-b79f-16026f903de9.png)
![Screenshot_20220205-193710](https://user-images.githubusercontent.com/80460567/152654836-f2b0907c-60bc-494d-9b11-862b1a545702.png)

https://ci.codeberg.org/qwerty287/Prism4j/build/6

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy).

**Comment by 6543** [comment]
Could be a gitea issue too ... Can you send us the related webhook. You can find it in giteas repo settings

**Comment by qwerty287** [comment]
```json
{
  "ref": "refs/heads/main",
  "before": "48f05bbb9cc473e5cbee258dc802079c6ef81795",
  "after": "5501febf8b28cf9c9101c2c11bb9d020a98d9ecf",
  "compare_url": "https://codeberg.org/qwerty287/wptest/compare/48f05bbb9cc473e5cbee258dc802079c6ef81795...5501febf8b28cf9c9101c2c11bb9d020a98d9ecf",
  "commits": [],
  "head_commit": {
    "id": "5501febf8b28cf9c9101c2c11bb9d020a98d9ecf",
    "message": "wp\n",
    "url": "https://codeberg.org/qwerty287/wptest/commit/5501febf8b28cf9c9101c2c11bb9d020a98d9ecf",
    "author": {
      "name": "qwerty287",
      "email": "ndev@web.de",
      "username": "qwerty287"
    },
    "committer": {
      "name": "qwerty287",
      "email": "ndev@web.de",
      "username": "qwerty287"
    },
    "verification": null,
    "timestamp": "2022-02-07T15:01:00+01:00",
    "added": [
      ".woodpecker.yml"
    ],
    "removed": [],
    "modified": []
  },
  "repository": {
    "id": 35952,
    "name": "wptest",
    "full_name": "qwerty287/wptest",
    "description": "",
    "empty": false,
    "private": false,
    "fork": false,
    "template": false,
    "parent": null,
    "mirror": false,
    "size": 26,
    "html_url": "https://codeberg.org/qwerty287/wptest",
    "ssh_url": "git@codeberg.org:qwerty287/wptest.git",
    "clone_url": "https://codeberg.org/qwerty287/wptest.git",
    "original_url": "",
    "website": "",
    "stars_count": 0,
    "forks_count": 0,
    "watchers_count": 1,
    "open_issues_count": 0,
    "open_pr_counter": 0,
    "release_counter": 0,
    "default_branch": "main",
    "archived": false,
    "created_at": "2022-02-07T14:59:45+01:00",
    "updated_at": "2022-02-07T15:01:38+01:00",
    "permissions": {
      "admin": true,
      "push": true,
      "pull": true
    },
    "has_issues": true,
    "internal_tracker": {
      "enable_time_tracker": true,
      "allow_only_contributors_to_track_time": true,
      "enable_issue_dependencies": true
    },
    "has_wiki": true,
    "has_pull_requests": true,
    "has_projects": true,
    "ignore_whitespace_conflicts": false,
    "allow_merge_commits": true,
    "allow_rebase": true,
    "allow_rebase_explicit": true,
    "allow_squash_merge": true,
    "default_merge_style": "merge",
    "avatar_url": "",
    "internal": false,
    "mirror_interval": ""
  }
  ...
}
```

`commits` is an empty list, but `head_commit` is provided and correct.

**Comment by qwerty287** [comment]
If I perform a force-push which adds new commits, everything's working. This only affects reset-only force-pushes (reset branch to an older commit and force-push it then)

**Comment by anbraten** [comment]
I am not quite sure, but could we simply ignore push webhooks without commits (empty array) or would it eat empty commits as well 🤔

**Comment by qwerty287** [comment]
Didn't test it, but empty commits should be in the `commits` array, why should they get removed?

**Comment by anbraten** [comment]
In that case the fix should be pretty simple I guess.

**Comment by qwerty287** [comment]
Only problem: if I use Woodpecker to publish build artifacts/binaries and I push a commit that contains an important bug, I'd force-push it with a reset (-> empty array). In this case I want Woodpecker to rebuild it.

**Comment by anbraten** [comment]
Should be fixed by #1150
